controller/linodecluster_controller_test.go

```diff
@@ -501,8 +501,18 @@ var _ = Describe("dns-override-endpoint", Ordered, Label("cluster", "dns-overrid
 			},
 		},
 	}
-	linodeMachines := infrav1alpha2.LinodeMachineList{
-		Items: []infrav1alpha2.LinodeMachine{linodeMachine},
+	linodeMachineWithNoAddress := infrav1alpha2.LinodeMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-machine",
+			Namespace: defaultNamespace,
+		},
+		Spec: infrav1alpha2.LinodeMachineSpec{
+			ProviderID: ptr.To("linode://123"),
+			InstanceID: ptr.To(123),
```

Comment:
Instance ID should be removed since it's deprecated

Reply:
yep - removed it